include/dlr_relayvm.h

```diff
@@ -0,0 +1,81 @@
+#ifndef DLR_RELAYVM_H_
+#define DLR_RELAYVM_H_
+
+#include <dlpack/dlpack.h>
+#include <tvm/runtime/container.h>
+#include <tvm/runtime/module.h>
+#include <tvm/runtime/ndarray.h>
+#include <tvm/runtime/object.h>
+#include <tvm/runtime/packed_func.h>
+#include <tvm/runtime/registry.h>
+#include <tvm/runtime/vm.h>
+#include "dlr_common.h"
+
+namespace dlr {
+
+
+/*! \brief class TVMModel
+ */
+class RelayVMModel : public DLRModel {
+ private:
+  std::vector<std::string> output_names_;
+  std::vector<std::string> output_types_;
+  nlohmann::json metadata_;
+  std::unique_ptr<ModelPath> path_;
+  tvm::runtime::ObjectPtr<tvm::runtime::vm::VirtualMachine> vm_;
+  std::shared_ptr<tvm::runtime::Module> vm_module_;
+  std::shared_ptr<tvm::runtime::Module> vm_executable_;
+  std::vector<tvm::runtime::NDArray> inputs_;
+  tvm::runtime::ObjectRef output_ref_;
+  std::vector<tvm::runtime::NDArray> outputs_;
+  void InitModelPath(std::vector<std::string> paths);
+  void SetupVMModule();
+  void FetchInputNodesData();
+  void LoadMetadata();
+  void FetchOutputNodesData();
+  void UpdateOutputs();
+  void UpdateInputs();
+  DLDataType GetInputDLDataType(int index);
+
+ public:
+  /*! \brief Load model files from given folder path.
+   */
+  explicit RelayVMModel(std::vector<std::string> paths, const DLContext& ctx)
+      : DLRModel(ctx, DLRBackend::kRELAYVM) {
+    InitModelPath(paths);
+    LoadMetadata();
+    SetupVMModule();
+    FetchInputNodesData();
+    FetchOutputNodesData();
+  }
+
+  int GetInputIndex(const char* name) const;
+  virtual const char* GetInputName(int index) const override;
+  virtual const char* GetInputType(int index) const override;
+  virtual const char* GetWeightName(int index) const override;
+  virtual std::vector<std::string> GetWeightNames() const override;
+  virtual void GetInput(const char* name, void* input) override;
+  virtual void SetInput(const char* name, const int64_t* shape, void* input,
+                        int dim) override;
+  virtual void Run() override;
+  tvm::runtime::NDArray GetOutput(int index);
+  virtual void GetOutput(int index, void* out) override;
+  virtual void GetOutputShape(int index, int64_t* shape) const override;
+  virtual void GetOutputSizeDim(int index, int64_t* size, int* dim) override;
+  virtual const char* GetOutputType(int index) const override;
+  virtual const char* GetBackend() const override;
+  virtual void SetNumThreads(int threads) override;
+  virtual void UseCPUAffinity(bool use) override;
+
+  /*
+    Following methods use metadata file to lookup input and output names.
```

Comment:
Should we move this to parent class since metadata file is uniform for both graph runtime and VM?

Reply:
There is a refactor PR where lot of common functionality is moved into parent class https://github.com/neo-ai/neo-ai-dlr/pull/213. Even realy vm changes will be ported over to that once merged.